### ISSUE TITLE
Why storing `type`s in `DataFrame`s is a Bad Idea

### ISSUE THREAD

**Issue by bengsparks** [pvs-hd-tea/PyTypes#127]
TL;DR: In order to unpickle `DataFrame`s with `type`s, every type stored therein must be imported prior to calling `pandas.read_pickle`.

Our initial implementation of the `Tracer` would store raw `type` objects into the `DataFrame`.
This was advantageous for us, as every instances' real type, location on disk for import-related tasks, and base class information were made available to anyone who loaded the `DataFrame`.
We had tested this on multiple basic types, such as `str` and `int`, then went on to trace a numpy test, which produced instances of `numpy.array`s. At this stage, the `DataFrame` STILL remained loadable.

Satisfied with this, we declared that we could build upon this `DataFrame` format, and began implementing further features on top of this. Then, when working on an MRE for inline annotation generation that would use class names as for the type hints, our `DataFrame` refused to load, stating that the given types could not be loaded.

As the TL;DR states states, this problem was fixed by importing the relevant class prior to unpickling.
However, this was not a viable solution for us.

The reason that even tracing the `numpy` test worked, is that when we imported `pandas` for unpickling, `numpy` was imported as a dependency of `pandas`, which made `numpy.array` accessible.

```py
λ python
Python 3.10.5 (main, Jun  6 2022, 18:49:26) [GCC 12.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas
>>> import sys
>>> sys.modules["numpy"]
<module 'numpy' from '/home/benji/.cache/pypoetry/
virtualenvs/solid-eureka-K39uNdS7-py3.10/lib/python3.10/site-packages/numpy/__init__.py'>
```

This was fixed by storing the module and name of the type as strings within the `DataFrame`, which is sufficient to dynamically import desired `type`s from the project, the standard library and from virtualenvs.
